src/test/javascript/spec/component/lecture-unit/online-unit/online-unit-form.component.spec.ts

```diff
@@ -0,0 +1,162 @@
+        onlineUnitFormComponentFixture.detectChanges();
+        expect(onlineUnitFormComponent.form.valid).toBeTrue();
+
+        const submitFormSpy = jest.spyOn(onlineUnitFormComponent, 'submitForm');
+        const submitFormEventSpy = jest.spyOn(onlineUnitFormComponent.formSubmitted, 'emit');
+
+        const submitButton = onlineUnitFormComponentFixture.debugElement.nativeElement.querySelector('#submitButton');
+        submitButton.click();
+
+        onlineUnitFormComponentFixture.whenStable().then(() => {
+            expect(submitFormSpy).toHaveBeenCalled();
+            expect(submitFormEventSpy).toHaveBeenCalledWith({
+                name: exampleName,
+                description: exampleDescription,
+                releaseDate: exampleReleaseDate,
+                source: 'https://www.example.com',
+            });
+
+            submitFormSpy.mockRestore();
+            submitFormEventSpy.mockRestore();
+        });
+    });
+
+    it('should correctly set form values in edit mode', () => {
+        onlineUnitFormComponent.isEditMode = true;
+        const formData: OnlineUnitFormData = {
+            name: 'test',
+            description: 'lorem ipsum',
+            releaseDate: dayjs().year(2010).month(3).date(5),
+            source: 'https://www.example.com',
+        };
+        onlineUnitFormComponentFixture.detectChanges();
+
+        onlineUnitFormComponent.formData = formData;
+        onlineUnitFormComponent.ngOnChanges();
+
+        expect(onlineUnitFormComponent.nameControl?.value).toEqual(formData.name);
+        expect(onlineUnitFormComponent.releaseDateControl?.value).toEqual(formData.releaseDate);
+        expect(onlineUnitFormComponent.descriptionControl?.value).toEqual(formData.description);
+        expect(onlineUnitFormComponent.sourceControl?.value).toEqual(formData.source);
+    });
+
+    it('should update when link changes', () => {
+        const onlineUnitService = TestBed.inject(OnlineUnitService);
+        const resourceDto = new OnlineResourceDTO();
+        resourceDto.title = 'Cum sociis natoque penatibus';
+        resourceDto.description = 'Pellentesque habitant morbi tristique senectus et netus.';
+        const response: HttpResponse<OnlineResourceDTO> = new HttpResponse({
+            body: resourceDto,
+            status: 200,
+        });
+        const getOnlineResourceStub = jest.spyOn(onlineUnitService, 'getOnlineResource').mockReturnValue(of(response));
+
+        onlineUnitFormComponentFixture.detectChanges();
+        onlineUnitFormComponent.isEditMode = true;
+        onlineUnitFormComponent.formData = {
+            source: 'example.com',
+        };
+        onlineUnitFormComponent.ngOnChanges();
+
+        // WHEN
+        onlineUnitFormComponent.onLinkChanged();
+
+        // THEN
+        expect(getOnlineResourceStub).toHaveBeenCalledOnce();
+        expect(onlineUnitFormComponent.sourceControl?.valid).toBe(true);
```

Comment:
You used ``toBeTrue()`` everywhere else :)
```suggestion
        expect(onlineUnitFormComponent.sourceControl?.valid).toBeTrue();
```